eth-registration/src/main/kotlin/registration/eth/relay/RelayRegistration.kt

```diff
@@ -69,6 +69,29 @@ class RelayRegistration(
         }
     }
 
+    /**
+     * Run a job that every replenishmentPeriod checks that number from config free relays are present. In case of
+     * lack of free relays deploys lacking amount.
+     */
+    fun runRelayReplenishment(): Result<Unit, Exception> {
+        logger.info { "Run relay replenishment" }
+
+        return Result.of {
+            while (true) {
```

Comment:
I think we can make it easier. Please, take a look at [ScheduledExecutorService](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html)

Reply:
I think it is not used for purpose since it is about async execution

Reply:
Do you mean `IrohaNetworkImpl`? I thought it's a thread-safe class